tests/components/deconz/test_sensor.py

```diff
@@ -289,16 +221,16 @@ async def test_add_battery_later(hass):
     data["sensors"] = {"1": deepcopy(SENSORS["3"])}
     gateway = await setup_deconz_integration(hass, get_state_response=data)
     remote = gateway.api.sensors["1"]
-    assert len(gateway.deconz_ids) == 0
+
+    assert len(hass.states.async_all()) == 0
     assert len(gateway.events) == 1
-    assert len(remote._callbacks) == 2
+    assert len(remote._callbacks) == 2  # Event and battery tracker
 
     remote.update({"config": {"battery": 50}})
     await hass.async_block_till_done()
 
-    assert len(gateway.deconz_ids) == 1
+    assert len(hass.states.async_all()) == 1
     assert len(gateway.events) == 1
-    assert len(remote._callbacks) == 2
+    assert len(remote._callbacks) == 2  # Event and battery entity
 
-    battery_sensor = hass.states.get("sensor.switch_1_battery_level")
-    assert battery_sensor is not None
+    assert hass.states.get("sensor.switch_1_battery_level")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6271139)